src/components/ArbeidsgiverPeriodeTabell.tsx

```diff
@@ -1,191 +1,136 @@
-import { connect } from 'react-redux';
-import { bindActionCreators, Dispatch } from 'redux';
-import { RootState } from '../store/rootState';
+import React, { useState } from 'react';
 import 'nav-frontend-tabell-style';
 import 'nav-frontend-skjema-style';
-import { ErrorType, Ytelsesperiode } from '../store/types/helseSpionTypes';
-import { Input } from 'nav-frontend-skjema';
-import { Søkeknapp } from 'nav-frontend-ikonknapper';
 import './ArbeidsgiverPeriodeTabell.less';
-import Lenke from 'nav-frontend-lenker';
-import { Innholdstittel } from 'nav-frontend-typografi';
 import 'nav-frontend-alertstriper-style';
 import 'react-datepicker/dist/react-datepicker.css';
-import { fetchPerson } from '../store/thunks/fetchPerson';
-import { identityNumberSeparation } from '../util/identityNumberSeparation';
-import AlertStripe from 'nav-frontend-alertstriper';
-import { withTranslation } from 'react-i18next';
-import { Keys } from '../locales/keys';
-import { filterStringToNumbersOnly } from '../util/filterStringToNumbersOnly';
-import YtelsesperiodeTable from './YtelsesperiodeTable';
-import { fetchArbeidsgivere } from '../store/thunks/fetchArbeidsgivere';
+import { useTranslation } from 'react-i18next';
 import Bedriftsmeny from '@navikt/bedriftsmeny';
 import '@navikt/bedriftsmeny/lib/bedriftsmeny.css';
-import { withRouter } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
 import { Organisasjon } from '@navikt/bedriftsmeny/lib/organisasjon';
+import { useAppStore } from '../data/store/AppStore';
+import { History } from 'history';
+import { Keys } from '../locales/keys';
+import Lenke from 'nav-frontend-lenker';
+import { Innholdstittel } from 'nav-frontend-typografi';
+import { identityNumberSeparation } from '../util/identityNumberSeparation';
+import { Input } from 'nav-frontend-skjema';
+import { Søkeknapp } from 'nav-frontend-ikonknapper';
+import { ErrorType } from '../store/types/helseSpionTypes';
+import AlertStripe from 'nav-frontend-alertstriper';
 import NavFrontendSpinner from 'nav-frontend-spinner';
+import YtelsesperiodeTable from './YtelsesperiodeTable';
+import Ytelsesperioder from './Ytelsesperioder';
 
-type OwnProps = {
-  t: (str: string) => string
-  history: History
-}
-
-type StateProps = {
-  arbeidsgivere: Organisasjon[]
-  ytelsesperioder: Ytelsesperiode[]
-  personErrorType?: string,
-  personErrorMessage?: string,
-  personLoading: boolean,
-}
-
-type DispatchProps = {
-  fetchPerson: (identityNumber?: string, arbeidsgiverId?: string) => void
-  fetchArbeidsgivere: () => void
-}
-
-type Props = OwnProps & StateProps & DispatchProps;
-
-type State = {
-  identityNumberInput: string
-  arbeidsgiverId: string
-}
-
-class ArbeidsgiverPeriodeTabell extends Component<Props, State> {
-  state: State = {
-    identityNumberInput: '',
-    arbeidsgiverId: '',
-  };
+const ArbeidsgiverPeriodeTabell: React.FC = () => {
+  const { arbeidsgivere } = useAppStore();
```

Comment:
Blir mange gjentakelser for "useAppStore" her. Kan det gjøres på annen måte?

Reply:
> Blir mange gjentakelser for "useAppStore" her. Kan det gjøres på annen måte?

Med redux :D